cmd/auth/token_test.go

```diff
@@ -0,0 +1,168 @@
+package auth_test
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"testing"
+	"time"
+
+	"github.com/databricks/cli/cmd"
+	"github.com/databricks/cli/libs/auth"
+	"github.com/databricks/cli/libs/auth/cache"
+	"github.com/databricks/cli/libs/databrickscfg/profile"
+	"github.com/databricks/databricks-sdk-go/httpclient"
+	"github.com/databricks/databricks-sdk-go/httpclient/fixtures"
+	"github.com/spf13/cobra"
+	"github.com/stretchr/testify/assert"
+	"golang.org/x/oauth2"
+)
+
+var refreshFailureTokenResponse = fixtures.HTTPFixture{
+	MatchAny: true,
+	Status:   401,
+	Response: map[string]string{
+		"error":             "invalid_request",
+		"error_description": "Refresh token is invalid",
+	},
+}
+
+var refreshFailureInvalidResponse = fixtures.HTTPFixture{
+	MatchAny: true,
+	Status:   401,
+	Response: "Not json",
+}
+
+var refreshFailureOtherError = fixtures.HTTPFixture{
+	MatchAny: true,
+	Status:   401,
+	Response: map[string]string{
+		"error":             "other_error",
+		"error_description": "Databricks is down",
+	},
+}
+
+var refreshSuccessTokenResponse = fixtures.HTTPFixture{
+	MatchAny: true,
+	Status:   200,
+	Response: map[string]string{
+		"access_token": "new-access-token",
+		"token_type":   "Bearer",
+		"expires_in":   "3600",
+	},
+}
+
+func validateToken(t *testing.T, resp string) {
+	res := map[string]string{}
+	err := json.Unmarshal([]byte(resp), &res)
+	assert.NoError(t, err)
+	assert.Equal(t, "new-access-token", res["access_token"])
+	assert.Equal(t, "Bearer", res["token_type"])
+}
+
+func getContextForTest(f fixtures.HTTPFixture) context.Context {
+	profiler := profile.InMemoryProfiler{
+		Profiles: profile.Profiles{
+			{
+				Name:      "expired",
+				Host:      "https://accounts.cloud.databricks.com",
+				AccountID: "expired",
+			},
+			{
+				Name:      "active",
+				Host:      "https://accounts.cloud.databricks.com",
+				AccountID: "active",
+			},
+		},
+	}
+	tokenCache := &cache.InMemoryTokenCache{
+		Tokens: map[string]*oauth2.Token{
+			"https://accounts.cloud.databricks.com/oidc/accounts/expired": {
+				RefreshToken: "expired",
+			},
+			"https://accounts.cloud.databricks.com/oidc/accounts/active": {
+				RefreshToken: "active",
+				Expiry:       time.Now().Add(1 * time.Hour), // Hopefully unit tests don't take an hour to run
+			},
+		},
+	}
+	client := httpclient.NewApiClient(httpclient.ClientConfig{
+		Transport: fixtures.SliceTransport{f},
+	})
+	ctx := profile.WithProfiler(context.Background(), profiler)
+	ctx = cache.WithTokenCache(ctx, tokenCache)
+	ctx = auth.WithApiClientForOAuth(ctx, client)
+	return ctx
+}
+
+func getCobraCmdForTest(f fixtures.HTTPFixture) (*cobra.Command, *bytes.Buffer) {
+	ctx := getContextForTest(f)
+	c := cmd.New(ctx)
+	output := &bytes.Buffer{}
+	c.SetOut(output)
+	return c, output
+}
```

Comment:
FYI there are helpers under `internal` that do the same and run the command in the background.

Reply:
Though you don't need interactive back-and-forth here, so this is also fine.